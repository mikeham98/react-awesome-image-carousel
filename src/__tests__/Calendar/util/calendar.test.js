import { getCalendarMonth, formatEvents, getEventsForCalendar, getMonthName } from '../../../app/ImageSlider/util/calendar';
import {events} from '../../__constants__/events';

describe('calendar', () => {
  describe('getCalendarMonth', () => {
    it('should return a month from 28th April - 1st June', () => {
      const month = 4;
      const year = 2019;
      expect(getCalendarMonth(month, year)).toMatchSnapshot();
    });
  });
  describe('formatEvents', () => {
    it('should return an object of key value pairs of dates with an array', () => {
      expect(formatEvents(events)).toMatchSnapshot();
    });
  });
});