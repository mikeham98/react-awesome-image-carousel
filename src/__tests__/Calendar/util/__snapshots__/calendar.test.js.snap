// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calendar formatEvents should return an object of key value pairs of dates with an array 1`] = `
Object {
  "1556665200000": Array [
    Object {
      "color": "#1ccb9e",
      "date": 2019-04-30T23:00:00.000Z,
      "from": "2019-05-01T13:00:00+00:00",
      "id": 2,
      "position": "startPosition",
      "spread": true,
      "title": "Holiday",
      "to": "2019-05-05T14:00:00+00:00",
    },
  ],
  "1556751600000": Array [
    Object {
      "color": "#1ccb9e",
      "date": 2019-05-01T23:00:00.000Z,
      "from": "2019-05-01T13:00:00+00:00",
      "id": 2,
      "position": "middlePosition",
      "spread": true,
      "title": "Holiday",
      "to": "2019-05-05T14:00:00+00:00",
    },
    Object {
      "color": "#fd3153",
      "date": 2019-05-01T23:00:00.000Z,
      "from": "2019-05-02T18:00:00+00:00",
      "id": 1,
      "position": "startPosition",
      "spread": true,
      "title": "Seeing family",
      "to": "2019-05-05T19:00:00+00:00",
    },
  ],
  "1556838000000": Array [
    Object {
      "color": "#1ccb9e",
      "date": 2019-05-02T23:00:00.000Z,
      "from": "2019-05-01T13:00:00+00:00",
      "id": 2,
      "position": "middlePosition",
      "spread": true,
      "title": "Holiday",
      "to": "2019-05-05T14:00:00+00:00",
    },
    Object {
      "color": "#fd3153",
      "date": 2019-05-02T23:00:00.000Z,
      "from": "2019-05-02T18:00:00+00:00",
      "id": 1,
      "position": "middlePosition",
      "spread": true,
      "title": "Seeing family",
      "to": "2019-05-05T19:00:00+00:00",
    },
  ],
  "1556924400000": Array [
    Object {
      "color": "#1ccb9e",
      "date": 2019-05-03T23:00:00.000Z,
      "from": "2019-05-01T13:00:00+00:00",
      "id": 2,
      "position": "middlePosition",
      "spread": true,
      "title": "Holiday",
      "to": "2019-05-05T14:00:00+00:00",
    },
    Object {
      "color": "#fd3153",
      "date": 2019-05-03T23:00:00.000Z,
      "from": "2019-05-02T18:00:00+00:00",
      "id": 1,
      "position": "middlePosition",
      "spread": true,
      "title": "Seeing family",
      "to": "2019-05-05T19:00:00+00:00",
    },
  ],
  "1557010800000": Array [
    Object {
      "color": "#1ccb9e",
      "date": 2019-05-04T23:00:00.000Z,
      "from": "2019-05-01T13:00:00+00:00",
      "id": 2,
      "position": "endPosition",
      "spread": true,
      "title": "Holiday",
      "to": "2019-05-05T14:00:00+00:00",
    },
    Object {
      "color": "#fd3153",
      "date": 2019-05-04T23:00:00.000Z,
      "from": "2019-05-02T18:00:00+00:00",
      "id": 1,
      "position": "endPosition",
      "spread": true,
      "title": "Seeing family",
      "to": "2019-05-05T19:00:00+00:00",
    },
    Object {
      "color": "#F480A8",
      "date": 2019-05-05T12:00:00.000Z,
      "from": 2019-05-05T12:00:00.000Z,
      "id": 3,
      "title": "Jet skiing",
      "to": 2019-05-05T14:00:00.000Z,
    },
    Object {
      "color": "#fda256",
      "date": 2019-05-05T18:00:00.000Z,
      "from": 2019-05-05T18:00:00.000Z,
      "id": 4,
      "title": "Dinner",
      "to": 2019-05-05T19:30:00.000Z,
    },
  ],
  "1557874800000": Array [
    Object {
      "color": "#8281fd",
      "date": 2019-05-15T12:00:00.000Z,
      "from": 2019-05-15T12:00:00.000Z,
      "id": 5,
      "title": "Doctors",
      "to": 2019-05-15T21:00:00.000Z,
    },
  ],
}
`;

exports[`calendar getCalendarMonth should return a month from 28th April - 1st June 1`] = `
Array [
  Array [
    Object {
      "current": false,
      "date": 2019-04-27T23:00:00.000Z,
    },
    Object {
      "current": false,
      "date": 2019-04-28T23:00:00.000Z,
    },
    Object {
      "current": false,
      "date": 2019-04-29T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-04-30T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-01T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-02T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-03T23:00:00.000Z,
    },
  ],
  Array [
    Object {
      "current": true,
      "date": 2019-05-04T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-05T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-06T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-07T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-08T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-09T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-10T23:00:00.000Z,
    },
  ],
  Array [
    Object {
      "current": true,
      "date": 2019-05-11T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-12T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-13T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-14T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-15T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-16T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-17T23:00:00.000Z,
    },
  ],
  Array [
    Object {
      "current": true,
      "date": 2019-05-18T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-19T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-20T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-21T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-22T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-23T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-24T23:00:00.000Z,
    },
  ],
  Array [
    Object {
      "current": true,
      "date": 2019-05-25T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-26T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-27T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-28T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-29T23:00:00.000Z,
    },
    Object {
      "current": true,
      "date": 2019-05-30T23:00:00.000Z,
    },
    Object {
      "current": false,
      "date": 2019-05-31T23:00:00.000Z,
    },
  ],
]
`;
